### ISSUE TITLE
On crash/reboot Container situation

### ISSUE THREAD

**Issue by ProJaCore** [pterodactyl/panel#1069]
* Version of Panel/Daemon: Both LATEST develop
* Server's OS: Ubuntu 16.04.3
* Your Computer's OS & Browser: FF-DE win10

------------------------

I experienced a crash and also did reboot my server:
after both actions i have following scenario:

The Servers are offline (they didnt start automatically).
I wasnt able to start the Servers. i got this following error:
Only after rebuilding the Container i was able to start the container.
![grafik](https://user-images.githubusercontent.com/12472791/37565356-5326c45a-2aa8-11e8-9777-dbdba17594af.png)
```
{"name":"wings","hostname":"srv01.host.com","pid":17335,"server":"1abef8de-f812-44c2-97f7-ea091a04a3d5","level":40,"err":{"message":"Unable to determine size of server data folder, exited with code 1 signal null.","name":"Error","stack":"Error: Unable to determine size of server data folder, exited with code 1 signal null.\n    at ChildProcess.Exec.on (/srv/daemon/src/controllers/fs.js:94:29)\n    at emitTwo (events.js:106:13)\n    at ChildProcess.emit (events.js:191:7)\n    at Process.ChildProcess._handle.onexit (internal/child_process.js:219:12)"},"msg":"Unable to determine size of server data folder, exited with code 1 signal null.","time":"2018-03-18T00:11:06.632Z","v":0}
{"name":"wings","hostname":"srv01.host.com","pid":5024,"level":30,"msg":"+ ------------------------------------ +","time":"2018-03-18T11:05:22.756Z","v":0}
{"name":"wings","hostname":"srv01.host.com","pid":5024,"level":30,"msg":"| Running Pterodactyl Daemon v0.0.0-canary    |","time":"2018-03-18T11:05:22.758Z","v":0}
{"name":"wings","hostname":"srv01.host.com","pid":5024,"level":30,"msg":"|        https://pterodactyl.io        |","time":"2018-03-18T11:05:22.758Z","v":0}
{"name":"wings","hostname":"srv01.host.com","pid":5024,"level":30,"msg":"|  Copyright 2015 - 2018 Dane Everitt  |","time":"2018-03-18T11:05:22.758Z","v":0}
{"name":"wings","hostname":"srv01.host.com","pid":5024,"level":30,"msg":"+ ------------------------------------ +","time":"2018-03-18T11:05:22.759Z","v":0}
{"name":"wings","hostname":"srv01.host.com","pid":5024,"level":30,"msg":"Loading modules, this could take a few seconds.","time":"2018-03-18T11:05:22.759Z","v":0}
{"name":"wings","hostname":"srv01.host.com","pid":5024,"level":30,"msg":"Modules loaded, starting Pterodactyl Daemon...","time":"2018-03-18T11:05:23.277Z","v":0}
{"name":"wings","hostname":"srv01.host.com","pid":5024,"level":30,"msg":"Configuring user pterodactyl (id: 118) as the owner of all server files.","time":"2018-03-18T11:05:23.337Z","v":0}
{"name":"wings","hostname":"srv01.host.com","pid":5024,"level":30,"msg":"Configuring timezone file location...","time":"2018-03-18T11:05:23.398Z","v":0}
{"name":"wings","hostname":"srv01.host.com","pid":5024,"level":30,"msg":"Checking container networking environment...","time":"2018-03-18T11:05:23.399Z","v":0}
{"name":"wings","hostname":"srv01.host.com","pid":5024,"level":30,"msg":"Found network interface for daemon: pterodactyl_nw","time":"2018-03-18T11:05:23.417Z","v":0}
{"name":"wings","hostname":"srv01.host.com","pid":5024,"level":30,"msg":"Ensuring correct network interface for containers...","time":"2018-03-18T11:05:23.417Z","v":0}
{"name":"wings","hostname":"srv01.host.com","pid":5024,"level":30,"msg":"Networking gateway detected as 172.18.0.1 for interface: pterodactyl0.","time":"2018-03-18T11:05:23.426Z","v":0}
{"name":"wings","hostname":"srv01.host.com","pid":5024,"level":30,"msg":"Contacting panel to retrieve a list of currrent Eggs available to the node.","time":"2018-03-18T11:05:23.428Z","v":0}
{"name":"wings","hostname":"srv01.host.com","pid":5024,"level":30,"msg":"Checking existing eggs aganist Panel response...","time":"2018-03-18T11:05:23.682Z","v":0}
{"name":"wings","hostname":"srv01.host.com","pid":5024,"level":30,"msg":"Beginning server initialization process.","time":"2018-03-18T11:05:23.752Z","v":0}
{"name":"wings","hostname":"srv01.host.com","pid":5024,"level":30,"msg":"Configuring websocket for daemon stats...","time":"2018-03-18T11:05:23.919Z","v":0}
{"name":"wings","hostname":"srv01.host.com","pid":5024,"level":30,"msg":"Configuring internal SFTP server...","time":"2018-03-18T11:05:23.919Z","v":0}
{"name":"wings","hostname":"srv01.host.com","pid":5024,"level":30,"msg":"Pterodactyl Daemon is up-to-date running a nightly build.","time":"2018-03-18T11:05:30.242Z","v":0}
{"name":"wings","hostname":"srv01.host.com","pid":5024,"level":30,"msg":"Pterodactyl Daemon is now listening for secure connections on 0.0.0.0:8700","time":"2018-03-18T11:05:30.243Z","v":0}
{"name":"wings","hostname":"srv01.host.com","pid":5024,"server":"1abef8de-f812-44c2-97f7-ea091a04a3d5","level":30,"msg":"Server status has been changed to STARTING","time":"2018-03-18T11:24:35.965Z","v":0}
{"name":"wings","hostname":"srv01.host.com","pid":5024,"server":"1abef8de-f812-44c2-97f7-ea091a04a3d5","level":30,"msg":"Server status has been changed to OFF","time":"2018-03-18T11:24:37.331Z","v":0}
{"name":"wings","hostname":"srv01.host.com","pid":5024,"server":"1abef8de-f812-44c2-97f7-ea091a04a3d5","level":50,"err":{"message":"(HTTP code 500) server error - container \"e56ca13dcfbee998c32ff4ffad2e4592e6b660084a96c4c5ade79cbba43d2e95\": already exists ","name":"Error","stack":"Error: (HTTP code 500) server error - container \"e56ca13dcfbee998c32ff4ffad2e4592e6b660084a96c4c5ade79cbba43d2e95\": already exists \n    at /srv/daemon/node_modules/docker-modem/lib/modem.js:254:17\n    at getCause (/srv/daemon/node_modules/docker-modem/lib/modem.js:284:7)\n    at Modem.buildPayload (/srv/daemon/node_modules/docker-modem/lib/modem.js:253:5)\n    at IncomingMessage.<anonymous> (/srv/daemon/node_modules/docker-modem/lib/modem.js:229:14)\n    at emitNone (events.js:91:20)\n    at IncomingMessage.emit (events.js:185:7)\n    at endReadableNT (_stream_readable.js:974:12)\n    at _combinedTickCallback (internal/process/next_tick.js:80:11)\n    at process._tickDomainCallback (internal/process/next_tick.js:128:9)"},"msg":"{ errorId: 'joH7HdkYRLaxTz9qumoB' }","time":"2018-03-18T11:24:37.335Z","v":0}
{"name":"wings","hostname":"srv01.host.com","pid":5024,"server":"1abef8de-f812-44c2-97f7-ea091a04a3d5","level":30,"msg":"Server status has been changed to STARTING","time":"2018-03-18T11:24:51.213Z","v":0}
{"name":"wings","hostname":"srv01.host.com","pid":5024,"server":"1abef8de-f812-44c2-97f7-ea091a04a3d5","level":30,"msg":"Server status has been changed to STOPPING","time":"2018-03-18T11:24:51.214Z","v":0}
{"name":"wings","hostname":"srv01.host.com","pid":5024,"level":30,"msg":"Checking if we need to update image quay.io/host/container:steam-updater, if so it will happen now.","time":"2018-03-18T11:24:51.373Z","v":0}
{"name":"wings","hostname":"srv01.host.com","pid":5024,"level":30,"msg":"Pulling image quay.io/host/container:steam-updater ... this could take a few minutes.","time":"2018-03-18T11:24:52.733Z","v":0}
{"name":"wings","hostname":"srv01.host.com","pid":5024,"server":"1abef8de-f812-44c2-97f7-ea091a04a3d5","level":30,"msg":"Completed rebuild process for server container.","time":"2018-03-18T11:24:54.818Z","v":0}
{"name":"wings","hostname":"srv01.host.com","pid":5024,"server":"1abef8de-f812-44c2-97f7-ea091a04a3d5","level":30,"msg":"Server status has been changed to OFF","time":"2018-03-18T11:24:54.821Z","v":0}
{"name":"wings","hostname":"srv01.host.com","pid":5024,"server":"1abef8de-f812-44c2-97f7-ea091a04a3d5","level":30,"msg":"Server status has been changed to STARTING","time":"2018-03-18T11:24:54.821Z","v":0}
{"name":"wings","hostname":"srv01.host.com","pid":5024,"server":"1abef8de-f812-44c2-97f7-ea091a04a3d5","level":30,"msg":"Server status has been changed to ON","time":"2018-03-18T11:24:58.124Z","v":0}
```
# This is not caused by my custom egg/docker image. please don't answer with a troubleshoot beside this topic.

**Comment by parkervcp** [comment]
It sounds like a docker issue honestly. It tried to start the failed container and it just needed to be removed and started over.

This is most likely not a panel/daemon issue. A crash is unexpected and cause issues like this.

**Comment by ProJaCore** [comment]
Can we add a automatic try to rebuild

**Comment by DaneEveritt** [comment]
I've not seen this occur enough to warrant digging into the logic surrounding that to handle issues like this. Theres also the issue of other problems looking _very_ similar to this, and a rebuild won't fix anything like it.

The container already exists issue shouldn't really happen anymore, and its been evading my figuring out the source for ages (but theres better logic to try and prevent that from happening).

Also, I apologize about the delay in responding to this.
